tests/build/multi-platform.go

```diff
@@ -51,218 +57,357 @@ var _ = framework.MultiPlatformBuildSuiteDescribe("Multi Platform Controller E2E
+			Expect(err).ShouldNot(HaveOccurred())
 		})
 
-		It("the build-container task from component pipelinerun is buildah-remote", func() {
+		When("the Component with multi-platform-build is created", func() {
+			It("a PipelineRun is triggered", func() {
+				Eventually(func() error {
+					pr, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(componentName, applicationName, testNamespace, "")
+					if err != nil {
+						GinkgoWriter.Printf("PipelineRun has not been created yet for the component %s/%s", testNamespace, componentName)
+						return err
+					}
+					if !pr.HasStarted() {
+						return fmt.Errorf("pipelinerun %s/%s hasn't started yet", pr.GetNamespace(), pr.GetName())
+					}
+					return nil
+				}, timeout, constants.PipelineRunPollingInterval).Should(Succeed(), fmt.Sprintf("timed out when waiting for the PipelineRun to start for the component %s/%s", testNamespace, componentName))
+			})
 
-			Eventually(func() error {
-				pr, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(componentName, applicationName, testNamespace, "")
-				Expect(err).ShouldNot(HaveOccurred())
+			It("the build-container task from component pipelinerun is buildah-remote", func() {
 
-				for _, chr := range pr.Status.ChildReferences {
-					taskRun := &pipeline.TaskRun{}
-					taskRunKey := types.NamespacedName{Namespace: pr.Namespace, Name: chr.Name}
-					err := f.AsKubeAdmin.CommonController.KubeRest().Get(context.TODO(), taskRunKey, taskRun)
+				Eventually(func() error {
+					pr, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(componentName, applicationName, testNamespace, "")
 					Expect(err).ShouldNot(HaveOccurred())
 
-					prTrStatus := &pipeline.PipelineRunTaskRunStatus{
-						PipelineTaskName: chr.PipelineTaskName,
-						Status:           &taskRun.Status,
-					}
+					for _, chr := range pr.Status.ChildReferences {
+						taskRun := &pipeline.TaskRun{}
+						taskRunKey := types.NamespacedName{Namespace: pr.Namespace, Name: chr.Name}
+						err := f.AsKubeAdmin.CommonController.KubeRest().Get(context.TODO(), taskRunKey, taskRun)
+						Expect(err).ShouldNot(HaveOccurred())
 
-					if chr.PipelineTaskName == constants.BuildTaskRunName && prTrStatus.Status != nil && prTrStatus.Status.TaskSpec != nil && prTrStatus.Status.TaskSpec.Volumes != nil {
-						for _, vol := range prTrStatus.Status.TaskSpec.Volumes {
-							if vol.Secret != nil && strings.HasPrefix(vol.Secret.SecretName, "multi-platform-ssh-") {
-								multiPlatformSecretName = vol.Secret.SecretName
-								return nil
+						prTrStatus := &pipeline.PipelineRunTaskRunStatus{
+							PipelineTaskName: chr.PipelineTaskName,
+							Status:           &taskRun.Status,
+						}
+
+						if chr.PipelineTaskName == constants.BuildTaskRunName && prTrStatus.Status != nil && prTrStatus.Status.TaskSpec != nil && prTrStatus.Status.TaskSpec.Volumes != nil {
```

Comment:
Shouldn't we wait/expect status for the task with the build name?
In fact that's what's done, but with a wrong error message.

Reply:
it was part of the previous existing test, IIUC, it is checking that the build-container task is using buildah-remote and not buildah, just by checking it has a secret attached as a volume, do you mean we should change the error message here to indicate that?

Reply:
IIUC, we want to check buildah-remote task and ensure that corresponding task run has a status, volume is set, etc. In my understanding we should wait for the conditions (it's done), however, if the task exists but volume is not set we'll get error `couldn't find a matching step buildah-remote` instead of missing volume for the task.
This is a nitpick, though.